### DISCUSSION TITLE
How to delete github copilot chat? Long github copilot chat crashes my Intellj

### DISCUSSION THREAD

**Discussion by JosipDomazetDev** [community/community#137257] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

I pasted a really long prompt into my github copilot chat. This chat causes the Intellj IDE to crash everytime I open the github copilot chat. How do I delete the chat? Uninstalling the plugin did not help, invalidating the cache did not help. It seems like the chat is stored online, other projects chats work fine.

Also note that I cannot access the UI for that chat at all and the settings in [github.com/settings/copilot](https://github.com/settings/copilot) don't seem to have that option.

Update/Solution: I had to clone the project from scratch unfortunately.

**Comment by Yackens** [comment]
It sounds like the problematic chat might be stored on GitHub's servers rather than locally in your IDE. Here’s how you might be able to resolve this issue:
1. Clear Chat Data: Check [GitHub's Copilot settings](https://github.com/settings/copilot) for options to clear chat history.
2. Check Online: Open GitHub Copilot Chat in a browser and see if you can delete the problematic chat.
3. Reset Settings: Try resetting your GitHub Copilot settings.
4. Contact Support: If needed, contact [GitHub Support](https://support.github.com/).
5. Reinstall IDE: As a last resort, consider reinstalling IntelliJ IDEA.

If you found this answer useful, please, mark my answer as the correct so other users can benefit from it.

**Comment by RodrigoDiasDeOliveira** [comment]
Potential Solutions
1. Clear Local Storage
Try clearing your browser's local storage for GitHub. This might help if the problematic chat is cached locally:

Open Developer Tools (F12 or Ctrl+Shift+I)
Go to the Application tab
Expand "Local Storage" in the left sidebar
Right-click on the GitHub domain and select "Clear"
2. Use GitHub CLI
You might be able to manage Copilot chats using the GitHub CLI. Install it if you haven't already, then try:

gh copilot chat list
gh copilot chat delete Copy
Replace with the ID of the problematic chat.

3. Contact GitHub Support
If the above solutions don't work, you'll need to contact GitHub support directly. They can access and manage your Copilot data on the server side.

[Contact GitHub Support](https://support.github.com/contact)
Prevention Tips
Avoid pasting extremely long prompts into Copilot chat
If you need to share large code snippets, consider using GitHub Gists instead
Regularly clear your Copilot chat history to prevent accumulation of problematic data

**Reply by JosipDomazetHKS** [reply]
Those tips unfortunately did not work and the issue still persists.

**Comment by tshandler** [comment]
1. Clear Copilot Data
Navigate the local Copilot configuration or data directory and try manually deleting the chat data.

2. Reinstall the Plugin
Uninstall the plugin through setting -->> Plugins
Reinstall again

3. Disable Problematic Chats (Optional)

**Reply by JosipDomazetHKS** [reply]
Deleting local data has no effect since its stored on the server :(

**Reply by tshandler** [reply]
If my approach is not working, try this again.

Open a New Chat Instance, instead of trying to access the problematic chat, start a new chat in GitHub Copilot with a different context or project.

**Reply by JosipDomazetHKS** [reply]
Other projects chats work fine but that particular one for that project always makes it crash. Since I cannot access that paritcular chat at all because it crashes my Intellj I cant interact with that chat at all. I can start the particular project no issues but when I open copilot chat it crashes my intellj.

**Reply by oriole9g** [reply]
Same problem :C any solution?

**Reply by JosipDomazetDev** [reply]
It doesnt work, I cloned the project again... Make sure to save all env files and so on.

**Reply by oriole9g** [reply]
An easy solution is simply rename the local project folder, delete /.idea folder and open the new folder as a new project.

**Comment by tshandler** [comment, marked as answer]
Have you ever created new project with your current source code?

**Reply by JosipDomazetHKS** [reply]
No I did not and this did in fact work. But for future readers: make sure to backup stuff like your .env files and co before you do this. Completely wiping your current project and cloning it from new should be a last ressort.

**Reply by niposch** [reply]
hitting: File > Invalidate Caches 
and selecting "Delete embedded browser engine cache and cookies" has worked for me. Rider will have to re-index the current solution though.

**Reply by ThoLyt** [reply]
Sorry about waking this question to life again, but I experienced a similar issue yesterday.

What worked for me was navigating to `/Users/[username]/.config/github-copilot/iu/`
There should be 3 folders there, named chat-agent-sessions, chat-edit-sessions and chat-sessions.
I deleted all files in the three folders and restarted my IDE. Now copilot is working again, but without any previous sessions

**Reply by melewetwo** [reply]
This worked for me on macos, thank you @ThoLyt

**Reply by theog** [reply]
Thanks @ThoLyt , This worked for me too. Using rubymine. `/Users/[username]/.config/github-copilot/rm/`

**Reply by maxmobilmaster-code** [reply]
GitHub Copilot Chat  v0.45.0 это дерьмо невозможно удалить заблокировать только выпилишь его а оно опять лезет все ломает .... суки

**Comment by nmz787-intel** [comment]
how to reset the "github copilot chat" in pycharm? the AI is stuck giving same ineffective responses and I want to clear all the context.

**Reply by nmz787-intel** [reply]
I asked AI how to do it... here's the working answer:
```
To reset the "GitHub Copilot Chat" in PyCharm, you can follow these steps:

Open the "GitHub Copilot Chat" panel in PyCharm.
Click on the three dots (menu) in the top-right corner of the chat panel.
Select "Reset Chat" or "Clear Context" from the dropdown menu.
This will clear the current context and reset the chat, allowing you to start fresh.
```

**Comment by oriole9g** [comment]
Same problem and none of these solutions work :( the only solution is to start a new project with the same code to effectively start a new copilot chat. Since the chat is on the copilot servers anything you do locally doesn't work because the plugin ends up syncing the chat and breaking everything when doing so.

**Reply by oriole9g** [reply]
When enable the plugin this try to sync the chat and always appears this error: 

`java.lang.IllegalArgumentException: Argument for @NotNull parameter 'fileType' of com/intellij/ide/highlighter/HighlighterFactory.createHighlighter` must not be null `

and there is no option to clear the chat because the plugin simply breaks when try to sync project chat.

**Comment by Cyclodex** [comment]
I had the issue in VSCode, but it might help maybe anyone with the same issue of github copilot chat history breaking the copilot completely.
* Solution: Check where your IDE stores chat history and delete those files.

For VSCode on Windows, it seems to be something like this:
```
%APPDATA%\Code\User\
```
there are folders like `History` (looks complex)...
For me it was good to look at `workspaceStorage` folder instead! I identified the correct workspace (check `workspace.json`), where there are folders like `chatSessions`, `chatEditiingSessions` and `GitHub.copilot-chat`.
In there, I deleted all history files from the day I know I introduced the bug into the chat history, this solved my problem (restarting editor after deleting those files).

So good luck in finding those files depending on your OS / setup.

```
C:\Users\*USERNAME*\AppData\Roaming\Code\User\workspaceStorage\*WORKSPACE*
```

**Reply by hotanphat28** [reply]
Thank you very much for sharing this details. it works for me! 🙏🏼

**Comment by CermakM** [comment]
In my case, I crashed Copilot chat in Goland by pasting a picture in it. :D Tried all the solutions above and none worked. I am on MacOS and deleting plugin cache folders didn't work either. The Copilot chat is unusable since then.

**Comment by mertsaner** [comment]
So, I have found the quick solution to this problem! Delete your .idea folder in your source code folder. Then copilot will be working like a charm again.

**Comment by luaugii** [comment]
Como todos aqui, tengo jetbrains  con el plugin de GitHub Copilot la version de paga. Un chat congela toda la maquina. Pero solamente es con un proyecto.

Que es lo que sí funciono:
1.- Movi el proyecto a otro dicrectorio.
2.- Borre toda la carpeta .idea (Utilizo Ubuntu desktop)
3.- Abri el PHPStorm y busque la carpeta del proyecto renombrado.

Solucionado.
Se borraron todos los Chats y no se congela el PHP STORM

**Comment by maxmanco** [comment]
I had the same issue. On Windows, all chat sessions are under `c:\Users\{username}\AppData\Local\github-copilot\iu\`. I deleted them, and Copilot stopped crashing.

**Comment by lanrehnics** [comment]
For Mac 
Delete everything in 
/Users/{your_user}/.config/github-copilot/ai/

**Comment by TheFoxy23** [comment]
## Deleting GitHub Copilot Chat in IntelliJ

GitHub Copilot chats in IntelliJ crash the IDE when they're corrupted by long prompts, and they're stored locally per project or IDE config rather than centrally on GitHub.

### Local Storage Paths
Manually delete chat data from these IDE-specific folders, then restart IntelliJ.

**macOS/Linux:**
```
~/.config/github-copilot/iu/
```
- Delete folders like `chat-agent-sessions`, `chat-edit-sessions`, `chat-sessions`.

**Windows:**
```
C:\Users\<username>\AppData\Local\github-copilot\<ide-code>\ (e.g., `iu` for IntelliJ Ultimate)
```
- Remove chat session files/folders.

**Project-level alternative:** Delete `.idea` folder in project root (backup first), then reimport project.

### Additional Troubleshooting Steps
- Invalidate caches: File > Invalidate Caches > Check "Delete embedded browser engine cache and cookies".
- Reinstall Copilot plugin after clearing data.
- Last resort: Clone project from scratch to reset synced state.

### Web vs. IDE Differences
- **Web (github.com):** Clear via browser storage or `gh copilot chat delete`.
- **IDE chats:** Local only; no GitHub settings option.

### VS Code Specifics
Delete from `%APPDATA%\Code\User\workspaceStorage\<hash>\chatSessions` folders.

If you still have any issue, or something doesn't work, obviously write here 🙃

**Comment by ShivendraSahu22** [comment]
To delete or clear your GitHub Copilot Chat history in IntelliJ IDEA and resolve performance issues, you can use the built-in commands or manually clear the local data storage. 
1. Clear Current Chat Responses
If you want to quickly clear the visible responses in the active chat window to reduce memory load, use the slash command:
Type /clear directly into the Copilot Chat input box and press Enter. 
Filo
Filo
 +2
2. Delete Specific Chat Sessions 
You can manage your history directly from the chat interface: 
Click the History icon (usually next to the "+" or under the menu lines) in the Copilot Chat window.
Find the session you want to remove.
Right-click on the session title and select Delete.
Tip: If you cannot delete the current active session, click the New Chat (+) button first to switch focus, then delete the old session from the history list. 
GitHub
GitHub
 +3
3. Manually Clear Local Storage (Recommended for Crashes) 
If long chat history is causing IntelliJ to crash or become unresponsive, you may need to manually delete the local cache files while the IDE is closed:
Windows: Navigate to C:\Users\<username>\AppData\Local\github-copilot\intellij and delete the contents.
macOS/Linux: Navigate to ~/.config/github-copilot/intellij and delete the stored chat data.
Project-Specific: Sometimes chat data is linked to the .idea folder in your project. Renaming your project folder and re-opening it as a "new" project can effectively clear that context. 
GitHub
GitHub
 +3
4. Additional Performance Fixes
If clearing the history doesn't stop the crashing, try these steps:
Update All Components: Ensure you are using the latest versions of IntelliJ IDEA and the GitHub Copilot plugin.
Invalidate Caches: Go to File > Invalidate Caches..., select all options, and click Invalidate and Restart.
Increase IDE Memory: Since long prompts cause high memory usage, go to Help > Change Memory Settings and increase the maximum heap size (e.g., to 4096MB or higher).

**Reply by vishalgir007** [reply]
You don’t necessarily need to re-clone the project—there are a few ways to recover from a broken Copilot chat session in IntelliJ IDEA.
